todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskActivity.java

```diff
@@ -80,20 +82,19 @@ protected void onCreate(Bundle savedInstanceState) {
             shouldLoadDataFromRepo = savedInstanceState.getBoolean(SHOULD_LOAD_DATA_FROM_REPO_KEY);
         }
 
-        // Create the presenter
-        mAddEditTaskPresenter = new AddEditTaskPresenter(
-                taskId,
+        mViewModel = new AddEditTaskViewModel(
+                getApplicationContext(),
```

Comment:
if we make it singleton, it would be just a Model (app specific) rather than a ViewModel (screen specific). I think we are fine just if we retain it.